packages/cdktf/lib/runtime.ts

```diff
@@ -48,6 +48,9 @@ export function hashMapper(elementMapper: Mapper): Mapper {
  * True unless it's undefined
  */
 export function canInspect(x: any) {
+  // Don't inspect IResolvable
+  if (x && "resolve" in x) return false;
```

Comment:
We should use `Tokenization.isResolvable(x)` here

Reply:
I believe I ran into issues there, but will check again.

Reply:
I still wonder if we should be doing a multi-phase resolve like I believe the AWS CDK is doing.

Reply:
@jsteinich me too - I haven't looked into this yet. Do you have more insights on reasons to do so?

Reply:
I think it would only matter if resolving one token resulting in a different token, but I really haven't dug into it either. Really more of a hunch that as we lean harder into the token system, we'll run into issues since we aren't doing things exactly the same way.

Reply:
See https://github.com/hashicorp/terraform-cdk/issues/910